src/reducers/SideBarReducer.js

```diff
@@ -1,16 +1,58 @@
-import { Map } from 'immutable';
+import { Map, List } from 'immutable';
 
 const INITIAL_STATE = Map({
-    currentSelectedCity: 'All cities',
+    'sideBarShown': true,
+    'currentSelectedCity': 'All cities',
+    'currentSelectedCategories': List(['All categories']),
+    'hasImages': false,
+    'newest': false
 });
 
 function SideBarReducer (state = INITIAL_STATE, action) {
     switch(action.type){
+    case('TOGGLE_SIDE_BAR'):
+        return state.set('sideBarShown', !state.get('sideBarShown'))
+    case('UPDATE_CATEGORY'):
+        return updateCategory(state, action);
     case('UPDATE_CITY'):
         return state.set('currentSelectedCity', action.data)
+    case('TOGGLE_HAS_IMAGES'):
+        return state.set('hasImages', !state.get('hasImages'))
+    case('TOGGLE_NEWEST'):
+        return state.set('newest', !state.get('newest'))
+    case('RESET_SELECTIONS'):
+        return state.set('currentSelectedCity', 'All cities').set('currentSelectedCategories', List(['All categories']));
     default:
         return state
     }
 }
 
+function updateCategory(state, action) {
+    //if all is selected again, then no changes to the category list
+    if (action.data === 'All categories') {
+        return state.set('currentSelectedCategories', List(['All categories']));
+    }
+
+    const currentStateCategoryList = state.get('currentSelectedCategories');
+
+    //if at first only 'all categories' in list,  erase 'all categories', then add new category to list 
+    if (currentStateCategoryList.get(0) === 'All categories' ) {
+        return state.set('currentSelectedCategories', List([action.data]))
+    }
+
+    const isCategorySelected = currentStateCategoryList.includes(action.data);
```

Comment:
This is good for readability! You don't necessarily need the variable assignment since you're about to use it right below, but it's beneficial for readability. +1

Reply:
I will keep it as it is then :)